cryostatio/cryostat-web
# fix(layout): fix content overflow

# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
_______________________________________________

Fixes: #828 

## Description of the change:

- [x] Enable horizontal scroll for rule table.
- [x] Collapse toolbar action buttons
  - [x] Active Recording Table
  - [x] Archive Recording Table. Note that PF advises against using overflow menu with only 2 buttons. So I add the breakpoint to lowest so that unless spaces are very limited, buttons are still shown.
  - [x]  Update tests. Got much complications with screen size testing.

## Motivation for the change:

The issue is quite severe for the Rule table as regular viewport also make it overflow. This PR addresses that and original issue with overflowing toolbar.

## How to manually test:
1. `Run CRYOSTAT_IMAGE=quay.io... sh smoketest.sh...`
2. Go to Automated Rule tab and create a rule if not any. See the table is not scrollable.
3. Go to Recording Tab. Reduce the screen viewport. See the buttons collapsing into a dropdowm


## Screenshots

**Collapsed toolbar:**

![Screenshot from 2023-02-02 16-02-45](https://user-images.githubusercontent.com/68053619/216448862-283eb55e-bf15-4fa3-82e7-552ccf4ea7bd.png)


**Collapsed Toolbar action buttons:**

![Screenshot from 2023-02-02 16-02-28](https://user-images.githubusercontent.com/68053619/216448868-1e1236a8-cdc8-4778-874c-1a706e8e0f32.png)

**Scrollable Automated table:**

![Screenshot from 2023-02-02 16-02-09](https://user-images.githubusercontent.com/68053619/216448870-f31f656a-2c00-4707-b721-0283d97c042e.png)
